### PR TITLE
Improve raw analysis and parallel delta loading

### PR DESCRIPTION
**Pull request by andreicuceu**
Improve the raw analysis by adding an option for linear binning when computing the deltas. Also add parallel loading of transmission files/writing of deltas (only for raw analysis). I also parallelised read_deltas in io.py, so everything that uses it should automatically read deltas in parallel. We need to test this and maybe add one more flag for the weight type of the raw deltas.

### CODE REVIEW

File: py/picca/converters.py
Comment (iprafols) [inline py/picca/converters.py:366]: This function should be in io.py

File: py/picca/converters.py
Comment (iprafols) [inline py/picca/converters.py:500]: This function should be in io.py

File: py/picca/io.py
Comment (iprafols) [inline py/picca/io.py:1478]: Removed commented version or add a comment specifying that it is the old way of loading data

File: py/picca/io.py
Comment (iprafols) [inline py/picca/io.py:1476]: This line is repeated (see line 1495). I would delete this here since there it also updates num_data

**Review by iprafols — CHANGES_REQUESTED** [review]
I'm not sure the linear binning will work with just the changes implemented here. We need to address at least also the functions in prep_del.py and the coadding and rebinning in data.py. There may be other places where changes are required as well, to be double-checked. 

That said, I would be happy to merge once the minor points I signalled are fixed. I think the parallelistion in the reading is long overdue

File: bin/picca_export.py
Comment (andreufont) [inline bin/picca_export.py:188]: Why is this change here? Has this leaked from your separate work on the blinding strategy?

**Comment by andreufont** [comment]
Thanks @iprafols , I think we can implement these minor changes.

Note, however, that the changes in this PR regarding log-lambda vs lin-lambda only affect the "deltas from transmission", so I don't think you have to worry about prep_del.py or the rebinning in data.py, since these are not involved in the script. We directly write delta files without any continuum fitting and without creating any Forest object.

File: bin/picca_export.py
Comment (andreicuceu) [reply bin/picca_export.py:188]: The version of master I started working from had a bug related to blinding (now fixed I think) and this is me doing a temporary fix so this branch works. Before merging I'll pull master into this and get rid of this line.

Comment (andreufont) [reply bin/picca_export.py:188]: Sounds good.

**Comment by andreufont** [comment]
@iprafols, @andreicuceu - now that we are done with the more pressing pull requests, it might be time to merge this one. We should also parallelise reading DESI data (it takes over an hour to read all spectra now in picca_deltas...), but we should do this in another PR.

**Comment by andreufont** [comment]
Yeah, I wasn't convinced about putting these to the (already too large) io.py. 

Putting these into raw_analysis.py or raw_io.py could be an option.

**Comment by iprafols** [comment]
Since they are io functions I would rather add them to a raw_io.py file than raw_analysis. 
I agree with @andreufont  that the io.py file is already too large

File: tutorials/desi/picca_convert_transmission.py
Comment (andreufont) [inline tutorials/desi/picca_convert_transmission.py:5]: Out of curiosity, why is this script in "tutorials/desi/"? I'd have imagine this would be in the main "/bins/" folder.

**Comment by andreicuceu** [comment]
Hi @iprafols and @andreufont I moved the functions to a new raw_io.py file, and updated an executable that runs the raw analysis, that is under tutorials for some reason. On my side I am happy to merge, but you can have another look to see if everything makes sense.

**Comment by andreufont** [comment]
I'd move the executable to /bin/ and merge.

**Review by iprafols — APPROVED** [review]

**Comment by iprafols** [comment]
It seems we are ready to merge, right?

**Comment by andreicuceu** [comment]
Yep

**Pull request merged**
